### ISSUE TITLE
[shelly] Support for Shelly 2PM Gen3 and Plus 2PM UL devices

### ISSUE THREAD

**Issue by weyto** [openhab/openhab-addons#17575]
## Feature Request

**Description**  
Currently, [Shelly 2PM Gen3 devices](https://www.shelly.com/products/shelly-2pm-gen3-1?srsltid=AfmBOorL92_TEVmDbm7YQV31SqxYKIARDtj7LQScb3dOseUQ2PHw0gOH) are being incorrectly discovered as `shelly:shellyunknown` in openHAB 4.2.2. This leads to a non functional device in openHAB.

I request an update to the Shelly binding in openHAB to correctly identify and support Shelly 2PM Gen3 devices.

### Environment:
- **Device Model**: [Shelly 2PM Gen3 Devices](https://www.shelly.com/products/shelly-2pm-gen3-1?srsltid=AfmBOorL92_TEVmDbm7YQV31SqxYKIARDtj7LQScb3dOseUQ2PHw0gOH)
- **Software Version**: openHAB 4.2.2

### Expected Behavior  
The system should properly recognize the Shelly 2PM Gen3 devices and expose all supported features such as power monitoring, relay controls, and any other relevant functionalities.

### Actual Behavior  
Shelly 2PM Gen3 devices are currently discovered as `shelly:shellyunknown`, resulting in the device not functioning as intended within openHAB.

**Comment by 3DJupp** [comment]
Hey - I have the same experience:
Unfortunately an old shelly 2.5 (3y+) was broken (faulty PSU) - so i would love to use it again with the replacement / Shelly 2 PM / Gen 3

**Comment by lsiepel** [comment]
Not sure, but I guess the exact name should be logged at debug level. Can you provide it?

**Comment by weyto** [comment]
@lsiepel Is this what you searching for?

`2024-11-07 08:07:18.137 [DEBUG] [discovery.ShellyDiscoveryParticipant] - shelly2pmg3-34cdb07721d8: Shelly device discovered: IP-Adress=192.168.2.43, type=shelly2pmg3
2024-11-07 08:07:18.185 [DEBUG] [g.shelly.internal.api2.Shelly2ApiRpc] - shelly2pmg3-34cdb07721d8: Connect Rpc Socket (discovery = true)
2024-11-07 08:07:18.188 [DEBUG] [helly.internal.api2.Shelly2RpcSocket] - shelly2pmg3-34cdb07721d8: Connect WebSocket, URI=ws://192.168.2.43/rpc
2024-11-07 08:07:18.366 [DEBUG] [helly.internal.api2.Shelly2RpcSocket] - shelly2pmg3-34cdb07721d8: WebSocket connected /192.168.2.100:36672<-/192.168.2.43:80, Idle Timeout=2147483647
2024-11-07 08:07:18.372 [DEBUG] [g.shelly.internal.api2.Shelly2ApiRpc] - shelly2pmg3-34cdb07721d8: Closing Rpc API (socket is connected, discovery=true)
2024-11-07 08:07:18.382 [DEBUG] [g.shelly.internal.api2.Shelly2ApiRpc] - shelly2pmg3-34cdb07721d8: Disconnect Rpc Socket
2024-11-07 08:07:18.386 [DEBUG] [helly.internal.api2.Shelly2RpcSocket] - shelly2pmg3-34cdb07721d8: Disconnecting WebSocket (/192.168.2.100:36672 -> /192.168.2.43:80)
2024-11-07 08:07:18.391 [DEBUG] [g.shelly.internal.api2.Shelly2ApiRpc] - shelly2pmg3-34cdb07721d8: WebSocket connection closed, status = 1006/Disconnected`

**Comment by lsiepel** [comment]
Briefly looking at the code and it seems like there is more that needs to be done and i lack knowledge of this specific binding devices to provide a good solution in a reasonable time.
With the provided information it should be doable.

**Comment by weyto** [comment]
@lsiepel : I looked into the history and found the commit fe0b1f8a201c03b4bca7b4f20121a140764b29cb by @markus7017. Is it not enought to extend [ShellyThingCreator.java](https://github.com/openhab/openhab-addons/blob/f0d4a0a965de293adbcc7f6b5e69443640a46bb0/bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyThingCreator.java) with the following changes?

` line 74 adding 2 new lines
public static final String SHELLYDT_PLUS2PMG3_RELAY = "S3SW-002P16EU-relay";
public static final String SHELLYDT_PLUS2PMG3_ROLLER = "S3SW-002P16EU-roller";
`

` line 385 add 2 new lines
THING_TYPE_MAPPING.put(SHELLYDT_PLUS2PMG3_RELAY, THING_TYPE_SHELLYPLUS2PM_RELAY_STR);
THING_TYPE_MAPPING.put(SHELLYDT_PLUS2PMG3_ROLLER, THING_TYPE_SHELLYPLUS2PM_ROLLER_STR);
`

**Comment by lsiepel** [comment]
Please try this jar: (openHAB 4.2.0+  compatible)
https://1drv.ms/u/s!AnMcxmvEeupwj4t06zw2TX4AzbuxWQ?e=CJL6FA

**Comment by lsiepel** [comment]
It is pretty safe to use in your production enviroment. If you uninstall the current binding and drop this jar into your addon folder, it won't affect things, channels and other settings.

**Comment by weyto** [comment]
The discovery is not working with the described changes above. It is more work to do.
In the ShellyThingCreator.java i found the getThingType method and there is a handling of the mode implemented. I'm not aware of all the things to do including all the dependencies.
Sorry

**Comment by weyto** [comment]
@markus7017 : Looks good for me: Device is discovered as shelly:shellyplus2pm-roller

**Comment by MatzeMuc86** [comment]
Any plan when to include to official release?
Has anyone tested 2 pm Gen 3 for rollers already?

**Comment by c214** [comment]
is there any snapshot-jar available makes shelly 2pm Gen.3 running?

**Comment by sydrvxd** [comment]
> is there any snapshot-jar available makes shelly 2pm Gen.3 running? I updated today to OH 4.3.2 but my shelly is still discovered as shellyunknown.

same with the latest version 4.3.3 :/

**Comment by Seestern93** [comment]
Hello together - I have 4.3.7 installed (latest for 32 bit OS -> could not update to 64 bit OS yet) but no support for my shelly2pmg3 -> shellyunknown

any news on that topic? or is it only fixed for OH 5?

**Comment by jlaur** [comment]
> any news on that topic? or is it only fixed for OH 5?

Yes, only in 5.0. If you follow the linked pull request, you can see that it is assigned to milestone 5.0.
